app/storage/encrypted_questionnaire_storage.py

```diff
@@ -26,11 +27,12 @@ def __init__(self, database, user_id, user_ik, pepper):
         self._cek = self._generate_key(user_id, user_ik, pepper)
 
     def add_or_update(self, data):
-        encrypted_data = self._encrypt_data(data)
+        serialised_data = simplejson.dumps(data, default=lambda o: o.__dict__)
```

Comment:
It's generally bad form to access an objects dunder (`__xx__`) methods and in this specific case I'm not sure it's a good idea. `__dict__` doesn't always exist for an object and it doesn't take account of an objects inheritance path so attributes that you'd expect to be serialised might not be.

Am I right that this is to cope with serialising Location objects? If it is then it might be preferable to define a public interface for getting a dict version of it *or* making it dict-like by inheriting from dict directly or [one of the collections abstract base classes](https://docs.python.org/3/library/collections.abc.html#collections.abc.Mapping).